plugin/metadata/provider.go

```diff
@@ -6,7 +6,14 @@
 //
 // Basic example:
 //
-// Implement the Provider interface for a plugin:
+// Implement the Provider interface for a plugin p:
+//
+//    func (p P) Metadata(ctx context.Context, state request.Request) context.Context {
+//       metadata.SetValueFunc(ctx, "test/something", func() string { return "myvalue" })
```

Comment:
Thanks for this - it helps clarify what's needed for https://github.com/coredns/coredns/pull/2682